javascript/test/basic_test.ts

```diff
@@ -801,4 +808,41 @@ describe("Automerge", () => {
       assert.deepEqual(path, pathCopy)
     })
   })
+
+  describe("independent creation of documents", () => {
+    // These tests document the behavior of Automerge when two peers create documents independently
+    // and then try to merge them.
+
+    it("merges non-conflicting fields", () => {
+      let doc1 = Automerge.init<any>()
+      let doc2 = Automerge.init<any>()
+      doc1 = Automerge.change(doc1, d => (d.x = 1))
+      doc2 = Automerge.change(doc2, d => (d.y = 2))
+      const merged = Automerge.merge(doc1, doc2)
+      // both properties are preserved
+      assert.deepEqual(merged, { x: 1, y: 2 })
+    })
+
+    it("merges non-conflicting fields, using Automerge.from", () => {
+      // shorthand for the above
+      let doc1 = Automerge.from<any>({ x: 1 })
+      let doc2 = Automerge.from<any>({ y: 2 })
+      const merged = Automerge.merge(doc1, doc2)
+      // both properties are preserved
+      assert.deepEqual(merged, { x: 1, y: 2 })
+    })
+
+    it("can't merge independently created arrays", () => {
```

Comment:
technically these are conflicts -- if you check there you'll see them.

Reply:
also @ept and @alexjg is it really wise for us to have a hardcoded object ID for the root?

Reply:
In my mind, whether the root object has a fixed object ID or not is an implementation detail; the real question is whether we want to allow merging of documents that were independently created and share no change history. If we wanted to rule out merging of independently created documents, a variable root ID that is created in the first change would be one way of achieving that, but a simpler solution would be to simply compare two docs' oldest change hash when merging, and refuse to merge if they don't match. But even if we restrict merges of independently created docs, that doesn't really solve the issue that Herb's tests here highlight, because the creation of the conflicting arrays need not be in the first change of the document's history, it could also happen later. The real solution, in my opinion, is to [merge concurrently created objects under the same key](https://github.com/automerge/automerge-classic/issues/4).